pkg/customresourcestate/config.go

```diff
@@ -158,21 +166,47 @@ type ConfigDecoder interface {
 	Decode(v interface{}) (err error)
 }
 
-// FromConfig decodes a configuration source into a slice of customresource.RegistryFactory that are ready to use.
-func FromConfig(decoder ConfigDecoder) ([]customresource.RegistryFactory, error) {
-	var crconfig Metrics
+// FromConfig decodes a configuration source into a slice of `customresource.RegistryFactory` that are ready to use.
+func FromConfig(decoder ConfigDecoder, gvkMaps *discovery.GVKMap) ([]customresource.RegistryFactory, error) {
```

Comment:
I think we should decouple the factory creation from this function.

This new feature highlights a pretty important gap which is that we don't support dynamic changes of the list of factories/stores. This mean for this feature that we would need to discover all the APIs before being able to run the server. If one API is missing at first but is present in the config (e.g: aggregated API not available yet when ksm start), we will never reevaluate the list of factories and include the missing API unless there is a new change to the config made that reloads the server.

Ideally we would want the discovery goroutine to trigger checks against the Resources extracted from the config and update the list of stores dynamically based on that to make sure that we are always up-to-date.